### ISSUE TITLE
100% cpu usages while using it with coc.nvim on mac

### ISSUE THREAD

**Issue by vivektiwary** [castwide/solargraph#187]
OS: macOs Mojave (version 10.14.5)
solargraph -  (0.32.3)
Ruby - 2.5.1

While using solargraph with coc.nvim in my neovim, the 
`2.5.1/bin/solargraph stdio` is taking 100% CPU.

**Comment by castwide** [comment]
Gem version 0.32.4 should resolve at least part of the problem. The language server's IO processing is much more efficient. (More info: castwide/vscode-solargraph#117)

Other than that, initialization can be intensive depending on the size of your workspace. It scans all of the available Ruby files, excluding a few directories like `spec`, to generate a map of all your code. You can tweak the map by creating a .solargraph.yml file (run `solargraph config`) and changing which files get included. (More info: https://github.com/castwide/vscode-solargraph/issues/116#issuecomment-496015930)

After initialization, the server process should idle at much lower usage. Let me know if you still have problems with it.

**Comment by kbarrette** [comment]
Just as an additional data point: I'm also seeing solargraph peg one of the cores on my MacBook.

Ruby: 2.6.2p47
solargraph: 0.34.2
coc.nvim: 0.0.72-8c7241d7e3

I haven't done much testing, but the problem seems to only occur after editing for a while - it doesn't start the moment I open a Ruby file in nvim.

Let me know if there's any data or logs you'd like me to collect!

**Comment by vivektiwary** [comment]
@kbarrette same thing is happening in my case as well. 100% cpu uses comes only after editing a file for sometime with approx 150 lines of code.

**Comment by castwide** [comment]
One thing that might help with debugging is running a scan in the workspace's root folder with `solargraph scan -v`. If it throws an exception or freezes on a particular node, it could help us determine the root cause. More information: https://solargraph.org/guides/scanning-workspaces

@vivektiwary Does it happen with any file, or just that one particular file? If the latter, I might be able to troubleshoot if you can c/p the code or reduce it to a minimal reproducible example.

**Comment by kbarrette** [comment]
```bash
$ solargraph scan -v
warning: parser/current is loading parser/ruby26, which recognizes
warning: 2.6.3-compliant syntax, but you are running 2.6.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
[WARN] Unable to load bibliothecary 6.7.3 specified by workspace, using 6.7.1 instead
...
[list of symbols]
...
Scanned /Users/[redacted] (21848 pins) in 22.768680999986827 seconds.
```

**Comment by vivektiwary** [comment]
@castwide There is no specific file, it happens for arbitrary files after some time. I normally work with more than 2 split buffer, and while switching as well the cpu usage increases.

**Comment by castwide** [comment]
Significant improvements have been made to performance since this issue was opened. If problems with CPU usage persist, please open a new issue and we'll try to pinpoint a root cause.
